### PR TITLE
Update to munit 1.0.0-M6

### PR DESCRIPTION
**Pull request by armanbilge**
Yoloing, let's see what CI says :)

### CODE REVIEW

File: munit/shared/src/main/scala/munit/ScalaCheckEffectSuite.scala [hunk 1]
```diff
@@ -96,8 +96,8 @@ trait ScalaCheckEffectSuite extends ScalaCheckSuite {
   // https://github.com/scalameta/munit/blob/68c2d13868baec9a77384f11f97505ecc0ce3eba/munit/shared/src/main/scala/munit/MUnitRunner.scala#L318-L326
   @tailrec
   private def rootCause(x: Throwable): Throwable = x match {
-    case _: InvocationTargetException | _: ExceptionInInitializerError |
-        _: UndeclaredThrowableException | _: ExecutionException if x.getCause != null =>
+    // should also include InvocationTargetException and UndeclaredThrowableException, but not on SJS
+    case _: ExceptionInInitializerError | _: ExecutionException if x.getCause != null =>
```
Comment (armanbilge) [inline munit/shared/src/main/scala/munit/ScalaCheckEffectSuite.scala:100]: These exceptions don't link for Scala.js. I'm not sure how much they matter. If anyone JVM-side complains we can always split this code cross-platform or something /shrug

Comment (armanbilge) [reply munit/shared/src/main/scala/munit/ScalaCheckEffectSuite.scala:99]: Note the PR that fixed this was observing an `ExecutionException`
- https://github.com/typelevel/scalacheck-effect/pull/196

File: munit/shared/src/main/scala/munit/ScalaCheckEffectSuite.scala [hunk 2]
```diff
@@ -85,23 +81,14 @@ trait ScalaCheckEffectSuite extends ScalaCheckSuite {
...
     )
 
-  // https://github.com/scalameta/munit/blob/68c2d13868baec9a77384f11f97505ecc0ce3eba/munit/shared/src/main/scala/munit/MUnitRunner.scala#L318-L326
-  @tailrec
-  private def rootCause(x: Throwable): Throwable = x match {
-    case _: InvocationTargetException | _: ExceptionInInitializerError |
-        _: UndeclaredThrowableException | _: ExecutionException if x.getCause != null =>
-      rootCause(x.getCause)
-    case _ => x
-  }
```
Comment (armanbilge) [inline munit/shared/src/main/scala/munit/ScalaCheckEffectSuite.scala:103]: This is now a public method in MUnit.
- https://github.com/scalameta/munit/pull/542

**Review by valencik — APPROVED** [review]
Looks good!
Yay deleted code :)

**Pull request merged**